src/pages/GoverningBody/Sector/AddSectorAdminForm.tsx

```diff
@@ -0,0 +1,318 @@
+    form.resetFields();
+    await NotificationBoxApi.createNotifications(
+      [admin.userId],
+      `Вам була присвоєна адміністративна роль: '${admin.adminType.adminTypeName}' в `,
+      NotificationBoxApi.NotificationTypes.UserNotifications,
+      `/governingBodies/${props.governingBodyId}`,
+      `цьому напрямі`
+    );
+  };
+
+  const editGoverningBodyAdmin = async (admin: SectorAdmin) => {
+    await editAdministrator(props.governingBodyId, admin);
+    notificationLogic("success", successfulEditAction("Адміністратора"));
+    form.resetFields();
+    await NotificationBoxApi.createNotifications(
+      [admin.userId],
+      `Вам була відредагована адміністративна роль: '${admin.adminType.adminTypeName}' в `,
+      NotificationBoxApi.NotificationTypes.UserNotifications,
+      `/governingBodies/${props.governingBodyId}`,
+      `цьому керівному органі`);
+  };
+
+
+  const showConfirm = (admin: SectorAdmin) => {
+    confirm({
+      title: "Призначити даного користувача на цю посаду?",
+      content: (
+        <div style={{ margin: 10 }}>
+          <b>
+            {head?.user.firstName} {head?.user.lastName}
+          </b>{" "}
+          є Головою Керівного Органу, час правління закінчується{" "}
+          <b>
+            {moment(head?.endDate).format("DD.MM.YYYY") === "Invalid date"
+              ? "ще не скоро"
+              : moment(head?.endDate).format("DD.MM.YYYY")}
+          </b>
+          .
+        </div>
+      ),
+      onCancel() { },
+      onOk() {
+        if (admin.id === 0) {
+          addGoverningBodyAdmin(admin);
+        } else {
+          editGoverningBodyAdmin(admin);
+        }
+      },
+    });
+  };
+
+  const handleSubmit = async (values: any) => {
+    const newAdmin: SectorAdmin = {
+      id: props.admin === undefined ? 0 : props.admin.id,
+      userId: props.admin === undefined
+        ? JSON.parse(values.userId).id
+        : props.admin.userId,
+      user: JSON.parse(values.userId),
+      adminType: {
+        ...new AdminType(),
+        adminTypeName: values.AdminType,
+      },
+      sectorId: props.sectorId,
+      startDate: values.startDate,
+      endDate: values.endDate,
+      workEmail: workEmail
+    };
+
+    onAdd();
+    if (newAdmin.id === 0) {
+      try {
+        if (values.AdminType === Roles.GoverningBodySectorHead && head !== null) {
+          if (head?.userId !== newAdmin.userId) {
+            showConfirm(newAdmin);
+          } else if (head?.userId === newAdmin.userId) {
+          } else {
+            editGoverningBodyAdmin(newAdmin);
+          }
+        } else {
+          if (newAdmin.id === 0) {
+            addGoverningBodyAdmin(newAdmin);
+          } else {
+            editGoverningBodyAdmin(newAdmin);
+          }
+        }
+      } finally {
+        onAdd();
+      }
+    }
+  };
+
+  const onUserSelect = (value: any) => {
+    const email: string = JSON.parse(value.toString()).email;
```

Comment:
Do you really need this variable?

Reply:
Yes, because i dont want to parse the value 3 times